ibis/backends/__init__.py

```diff
@@ -1116,13 +1116,20 @@ def _register_in_memory_tables(self, expr: ir.Expr) -> None:
         for memtable in expr.op().find(ops.InMemoryTable):
             if not self._in_memory_table_exists(memtable.name):
                 self._register_in_memory_table(memtable)
+                self._register_memtable_finalizer(memtable)
 
     def _register_in_memory_table(self, op: ops.InMemoryTable) -> None:
         if self.supports_in_memory_tables:
             raise NotImplementedError(
                 f"{self.name} must implement `_register_in_memory_table` to support in-memory tables"
             )
 
+    def _register_memtable_finalizer(self, op: ops.InMemoryTable) -> None:
```

Comment:
Having individual backends define their own method to create a finalizer feels a bit off to me. Could you instead define a method with a standard signature as a finalizer, like:

```python
def _finalize_memtable(self, name: str) -> None:
    ...
```

Then call `weakref.finalize` in `_register_in_memory_tables` uniformly for all backends? Should be effectively the same, and would keep all the weakref code in one spot.

Reply:
I think it's important the overridden methods be separate as opposed to together with the registration. This makes it easier to identify which backends need custom finalizers.

Reply:
Oh, I think I misunderstood.

Reply:
Yeah, I think I can do what you're suggesting.

Reply:
Done.